File: LibraryWeb/src/main/java/view/BookView.java
package view;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.*;

import javax.servlet.http.*;
import java.io.*;
import java.util.ArrayList;

/**
 * Created by Alex on 3/1/2018.
 */
public class BookView extends HttpServlet {

    @Override
    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try (PrintWriter out = response.getWriter()) {
            int id = (Integer)request.getAttribute("id");
            String title = request.getAttribute("title").toString();
            String author = request.getAttribute("author").toString();
            int year = (Integer)request.getAttribute("year");

            ObjectNode node = map.createObjectNode();
            node.put("id", id);
            node.put("title", title);
            node.put("author", author);
            node.put("year", year);
            array.add(node);
            out.write(array.toString());
        }
    }

    @Override
    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try(PrintWriter out = response.getWriter()) {
            boolean idStatus = (Boolean)request.getAttribute("IDStatus");
            ObjectNode node = map.createObjectNode();
            if(idStatus) {
                int id = (Integer)request.getAttribute("id");
                String title = request.getAttribute("title").toString();
                String author = request.getAttribute("author").toString();
                int year = (Integer)request.getAttribute("year");

                node.put("Status", "deleted");
                node.put("id", id);
                node.put("title", title);
                node.put("author", author);
                node.put("year", year);
            }else
                node.put("idError", "Unknown Book ID");
            array.add(node);
            out.write(array.toString());
        }
    }

    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try(PrintWriter out = response.getWriter()) {
            ArrayList<String> books = (ArrayList<String>)request.getAttribute("books");
            for (int i = 0; i < books.size(); i += 5) {
                ObjectNode node = map.createObjectNode();
                node.put("id", books.get(i));
                node.put("title", books.get(i+1));
                node.put("author", books.get(i+2));
                node.put("year", books.get(i+3));
                node.put("status", books.get(i+4));
                array.add(node);
            }
            out.write(array.toString());
        }
    }
}

File: LibraryWebSpark/src/main/java/view/BookView.kt
package view

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.node.*

/**
 * Created by Alex on 3/22/2018.
 */
object BookView {

    fun getBooks(books: ArrayList<String>, map: ObjectMapper): ArrayNode {
        val array = map.createArrayNode()
        var node: ObjectNode
        for (i in 0 until books.size step 5) {
            node = map.createObjectNode()
            array.add( node.run {
                put("id", books[i])
                put("title", books[i+1])
                put("author", books[i+2])
                put("year", books[i+3])
                put("status", books[i+4])
            } )
        }
        return array
    }

    fun addBook(id: Int, title: String, author: String, year: Int, map: ObjectMapper): ObjectNode {
        val node = map.createObjectNode()
        return node.run {
            put("id", id)
            put("title", title)
            put("author", author)
            put("year", year)
        }
    }

    fun deleteBook(id: Int, title: String, author: String, year: Int, map: ObjectMapper): ObjectNode {
        val node = map.createObjectNode()
        return node.run {
            put("status", "deleted")
            put("id", id)
            put("title", title)
            put("author", author)
            put("year", year)
        }
    }
}